tuya.cpp

```diff
@@ -474,422 +528,627 @@ void DeRestPluginPrivate::handleTuyaClusterIndication(const deCONZ::ApsDataIndic
+                                return;
+                            }
 
-                }
-                break;
-                case 0x0101: // off / running for Moe
-                {
-                    QString mode;
-                    if      (data == 0) { mode = QLatin1String("off"); }
-                    else if (data == 1) { mode = QLatin1String("heat"); }
-                    else
+                            if (item->toString() != mode)
+                            {
+                                update = true;
+                                
+                                item->setValue(mode);
+                                Event e(RSensors, RConfigPreset, sensorNode->id(), item);
+                                enqueueEvent(e);
+                            }
+                        }
+                    }
+                    break;
+                    case 0x0172: // Alarm siren humidity
                     {
-                        return;
+                        ResourceItem *item = sensorNode->item(RConfigPreset);
+                        
+                        if (item)
+                        {
+                            QString mode;
+                            if (data == 0)
+                            {
+                                if (item->toString() == "both")
+                                {
+                                    mode = QLatin1String("temperature");
+                                }
+                                else
+                                {
+                                    mode = QLatin1String("off");
+                                }
+                            }
+                            else if (data == 1)
+                            {
+                                if (item->toString() == "temperature")
+                                {
+                                    mode = QLatin1String("both");
+                                }
+                                else
+                                {
+                                    mode = QLatin1String("humidity");
+                                }
+                            }
+                            else
+                            {
+                                return;
+                            }
+
+                            if (item->toString() != mode)
+                            {
+                                update = true;
+                                
+                                item->setValue(mode);
+                                Event e(RSensors, RConfigPreset, sensorNode->id(), item);
+                                enqueueEvent(e);
+                            }
+                        }
                     }
+                    break;
+                    case 0x0269: // siren temperature
+                    {
+                        qint16 temp = (static_cast<qint16>(data & 0xFFFF)) * 10 + 200;
```

Comment:
```
qint16 temp = static_cast<qint16>(data & 0xFFFF) * 10 + 200;
```


Please remove all these extra parentheses around static cast they have no value other than adding visual clutter ;)

search for `(static_cast`  ....

Reply:
Have corrected mine.